roboteam_ai/src/bt/Role.cpp

```diff
@@ -40,10 +40,14 @@ Role::Role(std::string name) {
 }
 
 std::vector<Node::Ptr> Role::getChildren() {
-    return std::vector<Node::Ptr>{child};
+    if (child) {
```

Comment:
Is there a reason you changed this? It was actually kind of nice if this segfaulted because it was an easy way to tell your tree is invalid. Should atleast contain an error message IMO.

Reply:
wel... i got a segfault there

Reply:
So, there is a tree with a role with no child in it on this branch, which should never be the case (or your test has a role with no child in it), so I'd rather have it there. Our default behavior for tree structures not being right at the moment is AI just segfaulting on you so unless we are going to change that this atleast needs a print.